src/api_proxy/service_control/request_info.h

```diff
@@ -103,18 +103,19 @@ struct CheckRequestInfo : public OperationInfo {
   std::string ios_bundle_id;
 };
 
+enum CheckResponseErrorType {
+  UNKNOWN = 0,
+  API_KEY_INVALID = 1,
+  SERVICE_NOT_ACTIVATED = 2,
+  CONSUMER_BLOCKED = 3,
+  CONSUMER_ERROR = 4,
+};
+
 // Stores the information subtracted from the check response.
 struct CheckResponseInfo {
-  // If the request have a valid api key.
-  bool is_api_key_valid;
-  // If service is activated.
-  bool service_is_activated;
-  // Consumer project id
-  std::string consumer_project_id;
+  CheckResponseErrorType error_type = CheckResponseErrorType::UNKNOWN;
 
-  // By default api_key is valid and service is activated.
-  // They only set to false by the check response from server.
-  CheckResponseInfo() : is_api_key_valid(true), service_is_activated(true) {}
+  std::string consumer_project_id;
```

Comment:
keep the comment?

Reply:
Comment is no longer accurate / doesn't add any information